src/utils/espresso/sum_check/mod.rs

```diff
@@ -9,55 +9,51 @@
 
 //! This module implements the sum check protocol.
 
-use crate::utils::virtual_polynomial::{VPAuxInfo, VirtualPolynomial};
+use crate::{
+    transcript::Transcript,
+    utils::virtual_polynomial::{VPAuxInfo, VirtualPolynomial},
+};
+use ark_ec::{CurveGroup, Group};
 use ark_ff::PrimeField;
 use ark_poly::DenseMultilinearExtension;
 use ark_std::{end_timer, start_timer};
-use std::{fmt::Debug, sync::Arc};
+use std::{fmt::Debug, marker::PhantomData, sync::Arc};
 
-use espresso_subroutines::poly_iop::{prelude::PolyIOPErrors, PolyIOP};
-use espresso_transcript::IOPTranscript;
-use structs::{IOPProof, IOPProverState, IOPVerifierState};
+use crate::utils::sum_check::structs::IOPProverMessage;
+use crate::utils::sum_check::structs::IOPVerifierStateGeneric;
+use espresso_subroutines::poly_iop::prelude::PolyIOPErrors;
+use structs::{IOPProof, IOPProverState};
 
 mod prover;
 pub mod structs;
 pub mod verifier;
 
-/// Trait for doing sum check protocols.
-pub trait SumCheck<F: PrimeField> {
+/// A generic sum-check trait over a curve group
+pub trait SumCheckGeneric<C: CurveGroup> {
```

Comment:
Since the latest changes removed the need to differentiate between the original `SumCheck` and the generic `SumCheckGeneric` is no longer there, could we stay with the naming `SumCheck`? (removing the `Generic` word)
The reasoning is that the `Generic` is not differentiating anymore from a `non-generic` one, since the PR now adapts the original `SumCheck` to work with the `Transcript` trait.

(Same for the rest of `SumCheckVerifierGeneric` and `SumCheckVerifierStateGeneric`.)